pcdsdevices/attenuator.py

```diff
@@ -1071,6 +1072,114 @@ class AT2L0(FltMvInterface, PVPositionerPC, LightpathInOutMixin):
     num_out = Cpt(InternalSignal, kind='hinted')
+    
+    """check for errors and return a string indicating any errors verbally"""
+    def _check_errors(signals: SignalToValue) ->  str:
+        errors = []
+        #sort out .motor from .motor.plc signals
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value=sig.metadata.get('severity').value
+
+            if value not in (0,""):
+                errors.append(f"{sig.name}: {value}")
+
+        if errors:
+            return "\n".join(["Error summary:"] + errors)
+
+        return "No errors"
+
+    """summarize the errors occurring at any time on any blade via a string"""
+    error_summary = Cpt(
+        MultiDerivedSignalRO,
+        calculate=_check_errors,
+        attrs=sum(
+            (
+                [
+                    f"blade_{_blade:02d}.state.error",
+                    f"blade_{_blade:02d}.state.error_id",
+                    f"blade_{_blade:02d}.state.error_message",
+                    f"blade_{_blade:02d}.motor.plc.err_code",
+                    f"blade_{_blade:02d}.motor.user_readback",
+                ]
+                for _blade in range(1,20)
+            ),
+            [],
+        ),
+    )
+
+    """check for errors and return an array of binaries 1=error, 0=no error"""
+    def _check_errors_bitmask(signals: SignalToValue):
+        errors = []
+        blade_errors=[]
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value=sig.metadata.get('severity').value
+            
+            if value not in (0,""):
+                blade_errors.append(1)
+            else:
+                 blade_errors.append(0)
+
+        #remove error messages for blade one
+        del blade_errors[0:5] 
+        for idx in range(1,19):
+            #sum all errors on the 5 possible sources of error
+            if sum(blade_errors[0:5]) >= 1:
+                errors.append(1)
+            else:
+                errors.append(0)
+
+            del blade_errors[0:5]
+
+        decimal_value = 0
+        for next_bit in errors:
+            decimal_value = decimal_value * 2 + next_bit
+
+        return decimal_value
+    """summarize errors occurring at any time on any blade via a bitmask"""
+    error_summary_bitmask = Cpt(
+        MultiDerivedSignalRO,
+        calculate=_check_errors_bitmask,
+        attrs=sum(
+            (
+                [
+                    f"blade_{_blade:02d}.state.error",
+                    f"blade_{_blade:02d}.state.error_id",
+                    f"blade_{_blade:02d}.state.error_message",
+                    f"blade_{_blade:02d}.motor.plc.err_code",
+                    f"blade_{_blade:02d}.motor.user_readback",
+                ]
+                for _blade in range(1,20)
+            ),
+            [],
+        ),
+    )
+
+    def reset_err(self):
```

Comment:
Consider maybe a less similar to name to the signal so that the user won't get confused as easily, for example
```suggestion
    def clear_error(self):
```

Reply:
this change is probably for the best, it's also annoying switching between "err" and "error" just to avoid the signal going crazy.